perfkitbenchmarker/providers/gcp/util.py

```diff
@@ -13,26 +13,101 @@
 # limitations under the License.
 """Utilities for working with Google Cloud Platform resources."""
 
+from collections import OrderedDict
+
 from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
 
 FLAGS = flags.FLAGS
 
 
-def GetDefaultGcloudFlags(resource):
-  """Return common set of gcloud flags, including (format, project, zone).
-
-  Args:
-    resource: A GCE resource of type BaseResource.
+class GcloudCommand(object):
+  """A gcloud command.
 
-  Returns:
-    A common set of gcloud options.
+  Attributes:
+    args: list of strings. Non-flag args to pass to gcloud, typically
+        specifying an operation to perform (e.g. ['compute', 'images', 'list']
+        to list available images).
+    flags: OrderedDict mapping flag name string to flag value. Flags to pass to
+        gcloud (e.g. {'project': 'my-project-id'}). If a provided value is
+        True, the flag is passed to gcloud without a value. If a provided value
```

Comment:
It isn't common that there's a need for a program to support both a --flag and a --no-flag, since one of them will usually be the default behavior, so I don't think it's necessary to build that support in as a separate feature. This is more a matter of whether a flag has a value or not. I considered using None, rather than True, but it looks weird when you're setting options to explicitly provide a None value when you mean "enable this one". The way to set a --no-flag would be to pass {'no-flag': True} to this function.